flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptions.java

```diff
@@ -240,6 +235,18 @@
                                                     + "must be set to be greater than zero to enable sink buffer flushing.")
                                     .build());
 
+    public static final ConfigOption<DeliveryGuarantee> DELIVERY_GUARANTEE =
+            ConfigOptions.key("sink.delivery-guarantee")
+                    .enumType(DeliveryGuarantee.class)
+                    .defaultValue(DeliveryGuarantee.NONE)
+                    .withDescription("Optional delivery guarantee when committing.");
+
+    public static final ConfigOption<String> TRANSACTIONAL_ID_PREFIX =
+            ConfigOptions.key("sink.transactional-id-prefix")
+                    .stringType()
+                    .defaultValue(null)
+                    .withDescription("Optional delivery guarantee when committing.");
```

Comment:
c&p.

Reply:
please keep in mind that table users are even less skilled than DataStream API users when explaining this option. we have the infrastructure to add multiple paragraphs and code examples if it helps.